site/source/docs/api_reference/vr.h.rst

```diff
@@ -0,0 +1,378 @@
+	:rtype: bool
+
+.. c:function:: int emscripten_vr_get_display_capabilities(VRDisplayHandle handle, VRDisplayCapabilities* displayCaps)
+
+	:param VRDisplayHandle handle: |display-handle-parameter-doc|
+	:param VRDisplayCapabilities displayCaps: receives capabilities of the VR display.
+	:returns: |display-function-return-doc|
+	:rtype: bool
+
+.. c:function:: int emscripten_vr_get_eye_parameters(VRDisplayHandle handle, VREye whichEye, VREyeParameters* eyeParams)
+
+	:param VRDisplayHandle handle: |display-handle-parameter-doc|
+	:param VREye whichEye: which eye to query parameters for.
+	:param VREyeParameters eyeParam: receives the parameters for requested eye.
+	:returns: |display-function-return-doc|
+	:rtype: bool
+
+Render Loop
+===========
+
+In contrast to the usual emscripten main loop (see :ref:`emscripten-h-browser-execution-environment`),
+VR displays require their own rendering loop which is independent from the main loop. The rendering
+loop can be set per display and will act like a main loop with timing mode ``EM_TIMING_RAF`` until the
+display is requested to present, as of which it will run at the VR display's refresh rate.
+
+.. c:function:: void emscripten_vr_set_display_render_loop(VRDisplayHandle handle, em_vr_callback_func callback)
+
+	Set a C function as the per frame rendering callback of a VR display.
+
+	:param VRDisplayHandle handle: |display-handle-parameter-doc|: id of the display to set the render loop for.
+	:param em_vr_callback_func callback: C function to set as per frame rendering callback.
+	:rtype: |display-function-return-doc|
+
+.. tip:: There can be only *one* render loop function per VR display. To change the render loop function, first :c:func:`cancel <emscripten_vr_cancel_display_render_loop>` the current loop, and then call this function to set another.
+
+.. c:function:: void emscripten_vr_set_display_render_loop_arg(VRDisplayHandle handle, em_vr_callback_func callback, void* arg)
+
+	Set a C function as the per frame rendering callback of a VR display.
+
+	:param VRDisplayHandle handle: |display-handle-parameter-doc|
+	:param em_vr_callback_arg_func callback: C function to set as per frame rendering callback. The function signature must have a ``void*`` parameter for passing the ``arg`` value.
+	:param void* arg: User-defined data passed to the render loop function, untouched by the API itself.
+	:rtype: |display-function-return-doc|
+
+.. c:function:: void emscripten_vr_cancel_display_render_loop(VRDisplayHandle handle: |display-handle-parameter-doc|)
+
+	Cancels the render loop of a VR display should there be one running for it.
+
+	|render-loop-info|
+
+	:param VRDisplayHandle handle: |display-handle-parameter-doc|
+	:rtype: |display-function-return-doc|
+
+.. c:function:: int emscripten_vr_get_frame_data(VRDisplayHandle handle)
+
+	Get view matrix, projection matrix, timestamp and head pose for current frame.
+	Only valid when called from within a render loop callback.
```

Comment:
Meaning of return value is not documented here - is it a 0/1 success boolean?

Reply:
Also, based on documentation alone, for someone not aware about the internals, when this says that it gets the matrices, one can ask "where does it get them to?" since it does not return matrix values in this function signature, so that would be good to be documented as well.

Reply:
Right, will fix that!